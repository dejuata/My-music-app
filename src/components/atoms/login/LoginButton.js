import React from 'react';
import PropTypes from 'prop-types';

export const LoginButton = ({ url }) => {

    return (
        <a href={url} >
            <div className='btn btn-secondary'>
                Log In with Spotify
            </div>
        </a>
    )
}

LoginButton.propTypes = {
    url: PropTypes.string.isRequired
}
