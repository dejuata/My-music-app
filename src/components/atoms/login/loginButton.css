.a-login-button {
    background-color: var(--color-green);
    border-radius: 8px;
    border: none;
    width: 100%;
    cursor: pointer;
    font-size: var(--lg);
    font-weight: 500;
    text-align: center;
    padding: 20px;
    box-sizing: border-box;
}

.a-login-button a {
    text-decoration: none;
    color: var(--color-white);
}