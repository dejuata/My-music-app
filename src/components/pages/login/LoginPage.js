import React from 'react';

import { getUrlLogin } from '../../../api/selectors/getUrlLogin';
import { LoginButton } from '../../atoms/login/LoginButton';
import { Logo } from '../../atoms/Logo/Logo';
import { LoginForm } from '../../organisms/LoginForm/LoginForm';


export const LoginPage = () => {

    const loginUrl = getUrlLogin();

    return (
        <div className='auth-form__container'>
            <Logo />

            <LoginForm />

            <div className="auth__button">
                <LoginButton
                    url={ loginUrl }
                />
            </div>
        </div>
    )
}
