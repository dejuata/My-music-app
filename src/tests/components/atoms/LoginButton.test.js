import React from 'react';
import { shallow, mount } from 'enzyme';
import { LoginButton } from '../../../components/atoms/login/LoginButton';

describe('Test LoginButton', () => {

    const url = 'https://localhost:3000/url-authorization-spotify';

    const wrapper = shallow(
        <LoginButton url={ url }/>
    )


    test('Should be render LoginButton', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Should be have the login url', () => {
        const a = wrapper.find('a');

        expect( a.prop('href') ).toBe( url )
    })

})