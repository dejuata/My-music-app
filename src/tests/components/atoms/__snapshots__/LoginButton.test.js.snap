// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginButton Should be render LoginButton 1`] = `
<a
  href="https://localhost:3000/url-authorization-spotify"
>
  <div
    className="btn btn-secondary"
  >
    Log In with Spotify
  </div>
</a>
`;
